instance/tasks.py

```diff
@@ -81,13 +83,17 @@ def spawn_appserver(
 
 
 @db_task()
-def appserver_make_active(appserver_id, active=True):
+def appserver_make_active(appserver_id, active=True, deactivate_old=False):
     """
     Mark an AppServer as active or inactive.
     """
     logger.info('%s AppServer: ID=%s', "Activating" if active else "Deactivating", appserver_id)
     appserver = OpenEdXAppServer.objects.get(pk=appserver_id)
     appserver.make_active(active)
+    if deactivate_old:
+        for appserver_to_deactivate in appserver.instance.appserver_set.filter(_is_active=True):
+            logger.info(' - Deactivating %s [%s]', appserver, appserver.id)
+            appserver_to_deactivate.mark_active(active=False)
```

Comment:
I think this will trigger a LB reconfiguration twice.  The LB reconfiguration is a rather slow process (taking minutes), so it would be worth avoiding that.  I like Jill's suggestion to add a flag to `make_active()` to disable the LB reconfiguration.

Reply:
Yes, you're right. The flag suggestion makes sense, and I'm going to go down that route. Thanks!

Reply:
Could we instead make `make_active()` handle concurrent requests for reconfiguration gracefully? That would allow to avoid exposing its internals to functions that need to call it but not know what it does, like this, along with the mental overload that comes with it.

What about having the LB reconfiguration call turned into just flagging the LB config as modified, and needing a reconfiguration - and the actual reconfig could be done in a separate worker, either immediately via a task call if none is working, or once the previous reconfig is done if there is already one in progress?

Reply:
@antoviaque We have a ticket for doing exactly what you said in the backog, but we just don't have the time right now to implement this.  There are loads of things we want to improve in Ocim, and I don't feel the LB configurations are the highest priority right now.  If you feel it actually _is_ the top priority, we can start working on it, but that may delay our other plans.

My approach would be to keep it simple for now.  Adding the parameter Jill suggested now will simplify the code immediately, and we can further simplify it when we get to that ticket in the backlog.

Reply:
@smarnach  It's not the top priority as a task, but if we make a change on this, I'd rather take the clean approach now, rather than adding a new layer of complexity, which we will need to extricate when we would get to it later. It's exactly in this type of situation that we need to think about leaving the code cleaner after us, rather than dirtier.

And I assumed the change was relatively small - this is basically moving some of the code to another function/task, and gating the execution with a flag? 

For reference, this is https://tasks.opencraft.com/browse/OC-2517

Reply:
I don't mind pulling in OC-3157 and OC-2517 next sprint and making this PR basically solve both of them in one go, as they touch quite related parts/functionality. I don't intend on actually having this merged by the end of this week as part of OC-3242, because that wasn't actually planned (a small iteration was, but what I ended up doing, plus if we add OC-2517's work, wouldn't be) Does that sound good?

Reply:
@antoviaque Fair enough.  I don't really see the problem with the approach we planned, but it's not that important.  Implementing OC-2517 will work as well, so let's do that.  It won't resolve the double reconfiguration I mentioned above, but that's not a big problem – it will eventually disappear once we implement app server sets.

Reply:
@UmanShahzad @smarnach Sounds good - thanks to both of you!